thrift/server.cc

```diff
@@ -158,8 +161,23 @@ thrift_server::connection::process_one_request() {
         write().forward_to(std::move(_processor_promise));
         _processor_promise = promise<>();
     };
+    // Heuristics copied from transport/server.cc
+    size_t mem_estimate = 8000 + 2 * _input->available_read();
+    auto units = co_await get_units(_server._memory_available, mem_estimate);
+    // NOTICE: this permit is put in the server under the assumption that no other
+    // connection will overwrite this permit *until* it's extracted by the code
+    // which handles the Thrift request (via calling obtain_permit()).
+    // This assumption is true because there are no preemption points between this
+    // insertion and the call to obtain_permit(), which was verified both by
+    // code inspection and confirmed empirically by running manual tests.
+    if (_server._current_permit.count() > 0) {
+        tlogger.debug("Current service permit is overwritten while its units are still held ({}). "
```

Comment:
This is better than nothing, and I tend to agree to commit this, but I'm worried that maybe "debug" level is something nobody will ever notice :-(